### PR TITLE
docs(mongodb_metrics source): autogen cue docs

### LINKED ISSUE #15843: Convert the `mongodb_metrics` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by davidhuie-dd**
Closes #15843

### CODE REVIEW

File: src/sources/mongodb_metrics/mod.rs [hunk 1]
```diff
@@ -74,23 +75,27 @@ enum CollectError {
...
     #[serde(default = "default_scrape_interval_secs")]
-    scrape_interval_secs: u64,
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    scrape_interval_secs: Duration,
 
     /// Overrides the default namespace for the metrics emitted by the source.
     ///
     /// If set to an empty string, no namespace is added to the metrics.
     ///
+
```
Comment (neuronull) [inline src/sources/mongodb_metrics/mod.rs:98]: accident?

**Review by neuronull — COMMENTED** [review]
looks like some compilation errors

File: src/sources/mongodb_metrics/mod.rs [hunk 2]
```diff
@@ -1118,7 +1122,7 @@ mod integration_tests {
             tokio::spawn(async move {
                 MongoDbMetricsConfig {
                     endpoints,
-                    scrape_interval_secs: 15,
+                    scrape_interval_secs: Duration::from_secs(15),
```
Comment (spencergilbert) [inline src/sources/mongodb_metrics/mod.rs:1125]: 👀

**Review by spencergilbert — APPROVED** [review]

File: src/sources/mongodb_metrics/mod.rs (hunk 2 above)
Comment (neuronull) [reply src/sources/mongodb_metrics/mod.rs:1125]: I enabled integration tests for this PR... better safe than sorry :)

Comment (spencergilbert) [reply src/sources/mongodb_metrics/mod.rs:1125]: 😆

Comment (neuronull) [reply src/sources/mongodb_metrics/mod.rs:1125]: burned too many times :ded:

**Pull request merged**